test/dune

```
@@ -51,3 +51,39 @@
  (modules reg_45)
  (package printbox-text)
  (libraries printbox printbox-text))
+
+(test
+ (name extend)
+ (modules extend)
+ (package printbox-md)
```

```suggestion
 (package printbox-ext-plot)
```

only package to have all the dependencies I think

---

This test has no relation to the `printbox-ext-plot` package. This tests the extensibility mechanism. This test is the only test that tests extensibility for `printbox-md`, therefore it is attached to `printbox-md`. As a nice extra coincidence, `printbox-md` does have all the dependencies, but I think it's not a deciding factor (some other workaround would be required, e.g. removing the pieces that require a missing dependency).

---

my concern is that, when this is released on opam-repository, the `printbox-md` tests will be run in isolation. This test will be run in an environment that doesn't have `printbox-html` installed, for example, so it will fail.

---

I don't like relying on transitive dependencies, I'm OK with removing any mention of `PrintBox_html` and `PrintBox_text` from this test -- make it a purely `printbox-md` test. I'll do that. But `printbox-md` depends on `printbox-html`.

---

Very offtopic, but:

One could argue that it's a bad design choice for `printbox-md` to depend on `printbox-html`, and that instead `printbox-md` should offer either:
- only the partial coverage of the representation that doesn't rely on fallbacks, and not have extra dependencies,
- only the textual fallback (no HTML fallback).

I'm actually OK with changing `printbox-md` in either of these two directions, after experiencing that Markdown renderers are surprisingly inefficient with big complex files, i.e. discovering that my motivating use-case is impractical.

---

Note it is only conceptually a transitive dependency on `printbox-html`, technically it's a direct dependency.

---

Done with next commit.

---

ok, just removing the `printbox-text` case and keeping it in `printbox-md` is good enough for me.

I don't have strong opinions on how to change printbox-md (if you want to) but I'd say it's probably good enough to only keep the textual output since it's what I'd personally expect from the label on the tin. :)

---

> just removing the `printbox-text` case

I assume you misspoke and mean removing the `printbox-html` case. I removed both so it only tests the Markdown backend.